uk-climate-analysis/README.md

```diff
@@ -25,13 +25,61 @@ datasets needed to run the model for all feature types are listed below:
 
 • https://data.ceda.ac.uk/badc/ukcp09/data/gridded-land-obs/gridded-land-obs-monthly/grid/ascii/sunshine
 
+Note, these data have already been loaded into DAFNI, so with the appropriate permissions (just ask us) you can access them directly there to save space. Alternatively download the files from CEDA and file them in your folder structure (noting each of these is a zip file, thus in /data/inputs/rainfall you have 'rainfall.zip' etc.) The data will then be incorporate into the dockr image that you upload.
+
+## Creating DOCKER container
+After installing Docker on your computer, below are the commands you will need to run to create your docker image of the model ready to upload to DAFNI.
+See https://docs.secure.dafni.rl.ac.uk/docs/how-to/models/how-to-create-a-dafni-ready-model/
+
+### Build container
+`docker build -t uk-climate-analysis .`
+
+### Run container
+`docker run uk-climate-analysis`
+
+### Build image
+`docker save -o uk-climate-analysis.tar uk-climate-analysis`
+
+### Compress resultant tar image
+`gzip uk-climate-analysis.tar`
+This compresse gz file, and the yaml file can now be uploaded to DAFNI.
+
 ## Running the Model
+Running the model in DAFNI follows a different path from the text below - the DAFNI `Dockerfile` contains the following two lines:
+
+`RUN pip install -r requirements.txt`<br />
+`CMD python climate_analysis.py`
+
+So in DAFNI the model file (climate_analysis.py) is run automatically. However, for reference the text below explains how the model could be run on your computer stand-alone.
+
 To run the model, first install the requirements with `pip install -r
 requirements.txt`. Then, change the `OUTPUT_FOLDER` and `INPUT_FOLDER` in the
-`climate_analysis.py` file to match your actual data folders. 
+`climate_analysis.py` file to match your actual data folders.
 
-Then, use `python climate_analysis.py` to run the model. 
+Then, use `python climate_analysis.py` to run the model.
 
+## Visualising the Model Output
 The `visualisation.py` file assumes you have imported the data into a Jupyter
 notebook but you don't have to do this. You can adapt the `get_output_folder()`
-folder to point to your output folder instead.
\ No newline at end of file
+folder to point to your output folder instead.
+
+In DAFNI, you will create a new 'Jupyter notebook' in the visualisation tool and then copy and paste in the ode from the `visualisation.py` file.
```

Comment:
```suggestion
In DAFNI, you will create a new 'Jupyter notebook' in the visualisation tool and then copy and paste in the code from the `visualisation.py` file.
```

Reply:
Edits made to source and saved back - plus other typos spotted!